apps/rush-lib/src/cli/actions/AddAction.ts

```diff
@@ -0,0 +1,115 @@
+      + ' the ensureConsistentVersions policy). If a version range is specified, the latest version'
+      + ' in the range will be used. The version will be automatically prepended with a tilde, unless'
+      + ' the --exact or --caret flags are used. The --make-consistent flag can be used to update'
+      + ' all packages with the dependency.'
+    ];
+    super({
+      actionName: 'add',
+      summary: 'Adds a dependency to the package.json and runs rush upgrade.',
+      documentation: documentation.join(os.EOL),
+      safeForSimultaneousRushProcesses: false,
+      parser
+    });
+  }
+
+  public onDefineParameters(): void {
+    this._packageName = this.defineStringParameter({
+      parameterLongName: '--package',
+      parameterShortName: '-p',
+      required: true,
+      argumentName: 'PACKAGE_NAME',
+      description: '(Required) The name of the package which should be added as a dependency'
+    });
+    this._versionSpecifier = this.defineStringParameter({
+      parameterLongName: '--version',
+      parameterShortName: '-v',
+      argumentName: 'VERSION_RANGE',
+      description: 'An optional version specifier. If specified, the largest version satisfying this range'
+        + ' will be added to the package.json.'
+    });
+    this._exactFlag = this.defineFlagParameter({
+      parameterLongName: '--exact',
+      parameterShortName: '-e',
+      description: 'If specified, the SemVer specifier added to the'
+        + ' package.json will be a locked, exact version.'
+    });
+    this._caretFlag = this.defineFlagParameter({
+      parameterLongName: '--caret',
+      parameterShortName: '-c',
+      description: 'If specified, the SemVer specifier added to the'
+        + ' package.json will be a prepended with a "caret" specifier ("^").'
+    });
+    this._devDependencyFlag = this.defineFlagParameter({
+      parameterLongName: '--dev',
+      parameterShortName: '-d',
+      description: 'If specified, the package will be added as a "devDependency"'
+        + ' to the package.json'
+    });
+    this._makeConsistentFlag = this.defineFlagParameter({
+      parameterLongName: '--make-consistent',
+      parameterShortName: '-m',
+      description: 'If specified, other packages with this dependency will have their package.json'
+        + ' files updated to use the same version of the dependency.'
+    });
+    this._skipUpdateFlag = this.defineFlagParameter({
+      parameterLongName: '--skip-update',
+      parameterShortName: '-s',
+      description: 'If specified, the "rush update" command will not be run after updating the'
+        + ' package.json files.'
+    });
+  }
+
+  public run(): Promise<void> {
+    const project: RushConfigurationProject | undefined
+      = this.rushConfiguration.getProjectForPath(process.cwd());
+
+    if (!project) {
+      return Promise.reject(new Error('Not currently in a project folder'));
```

Comment:
>Not currently [](start = 39, length = 13)

The "rush add" command must be invoked under a project folder that is registered in rush.json

Reply:
We could relax this in the future

---
In reply to: [220754019](https://github.com/Microsoft/web-build-tools/pull/843#discussion_r220754019) [](ancestors = 220754019)